internal/cmddiff/cmddiff_test.go

```diff
@@ -69,3 +71,29 @@ func TestCmdExecute(t *testing.T) {
 	err = runner.C.Execute()
 	assert.NoError(t, err)
 }
+
+func TestCmd_flagAndArgParsing_Symlink(t *testing.T) {
+	dir, err := ioutil.TempDir("", "")
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+	defer os.RemoveAll(dir)
+	defer testutil.Chdir(t, dir)()
+
+	err = os.MkdirAll(filepath.Join(dir, "path", "to", "pkg", "dir"), 0700)
+	assert.NoError(t, err)
+	err = os.Symlink(filepath.Join("path", "to", "pkg", "dir"), "foo")
+	assert.NoError(t, err)
+
+	// verify the branch ref is set to the correct value
+	r := cmddiff.NewRunner(fake.CtxWithDefaultPrinter(), "kpt")
+	r.C.RunE = NoOpRunE
+	r.C.SetArgs([]string{"foo" + "@refs/heads/foo"})
+	err = r.C.Execute()
+	assert.NoError(t, err)
+	cwd, err := os.Getwd()
+	assert.NoError(t, err)
+	assert.Equal(t, filepath.Join(cwd, "path", "to", "pkg", "dir"), r.Path)
+}
```

Comment:
I am skeptical of tests that verifies if internal variables of a command have been set up correctly. They provide little utility (because they don't test the command behavior but the implementation detail) and results in churn whenever command implementation changes. 
Haven't thought a lot about it, is there a way to capture symlinks case in e2e ?

Reply:
This is an e2e test with Noop RunE. This is the general pattern we are following to test the pkg commands path related constructs.

Reply:
I know we have been following that pattern. My comment was about challenging that pattern itself. nvm, don't think its a blocker for the PR, but wanted to let you know my opinion about that pattern :)